test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpFixtureWithSTS.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package fixture.s3;
+
+import com.sun.net.httpserver.HttpHandler;
+
+import org.elasticsearch.rest.RestStatus;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class S3HttpFixtureWithSTS extends S3HttpFixture {
+
+    private static final String WEB_IDENTITY_TOKEN =
```

Comment:
Do you think we can find a way to avoid duplicating the token here and in the test file? Can we pass the path to the test file to the fixture, maybe?

Reply:
Removed the token from here, and moved it to `docker-compose` with a custom Dockerfile accepting a web idenity token.
The token is still hardcoded in the docker compose file, but at least the fixture is reusable.